tests/annotation_model_tests/test_annotation_model_nieuwkoopse_plassen.py

```diff
@@ -0,0 +1,192 @@
+import io
+import sys
+
+# Adjust this for the tests!!
+model_path = "C:/repos/satellite-images-nso-datascience/saved_models/Superview_Nieuwkoopse_plassen_20190302_113613_to_20221012_104900_random_forest_classifier.sav"
+location = "Nieuwkoopse_plassen"
+satellite_constellation = "Superview"
+output_path_test = "E:/output/test/Nieuwkoopse_plassen/"
+output_path = "E:/output/test/Nieuwkoopse_plassen/"
+
+# Put in settings file
+test_tif_files_dir = "E:/output/test/Nieuwkoopse_plassen/*SV*.tif"
+final_artefact = pickle.load(open(model_path, "rb"))
+
+
+def raster_to_dataframe(a_tif_file):
+    src = rasterio.open(a_tif_file)
+    data = src.read()
+    z_shape = data.shape[0]
+    x_shape = data.shape[1]
+    y_shape = data.shape[2]
+
+    x_coordinates = [
+        [x for y in range(0, data.shape[2])] for x in range(0, data.shape[1])
+    ]
+    y_coordinates = [
+        [y for y in range(0, data.shape[2])] for x in range(0, data.shape[1])
+    ]
+
+    rd_x, rd_y = rasterio.transform.xy(src.transform, x_coordinates, y_coordinates)
+
+    data = np.append(data, rd_x).reshape([z_shape + 1, x_shape, y_shape])
+    data = np.append(data, rd_y).reshape([z_shape + 2, x_shape, y_shape])
+
+    data = data.reshape(-1, x_shape * y_shape).transpose()
+
+    df = pd.DataFrame(
+        data,
+        columns=selected_features + ["rd_x", "rd_y"],
+    )
+
+    return df[(df[["r", "g", "b"]] != 0).any(axis="columns")]
+
+
+# Init model locations
+if location == "Voornes Duin":
+    selected_features = ["r", "g", "b", "n", "e", "d", "ndvi", "re_ndvi", "height"]
+
+
+elif location == "Coepelduynen":
+    # Optimal parameters and features
+    selected_features = ["r", "g", "b", "n", "e", "d", "ndvi", "re_ndvi"]
+
+
+elif location == "Schippersgat" and satellite_constellation == "Superview":
+    # Optimal parameters and features
+    selected_features = ["r", "g", "b", "i", "ndvi", "height"]
+
+
+elif location == "Schippersgat" and satellite_constellation == "PNEO":
+    # Optimal parameters and features
+    selected_features = ["r", "g", "b", "n", "e", "d", "ndvi", "re_ndvi"]
+
+
+elif location == "Nieuwkoopse_plassen" and satellite_constellation == "PNEO":
+    # Optimal parameters and features
+    selected_features = ["r", "g", "b", "n", "e", "d", "ndvi", "re_ndvi"]
+
+
+elif location == "Nieuwkoopse_plassen" and satellite_constellation == "Superview":
+    # Optimal parameters and features
+    selected_features = ["r", "g", "b", "i", "ndvi", "ndwi"]
+
+
+def test_difficult_pixels():
+
+    if location == "Nieuwkoops_plassen" and satellite_constellation == "Superview":
+        assert (
+            final_artefact["model"].predict(
+                final_artefact["scaler"].transform([[275, 208, 117, 219, 97, 88]])
+            )
+            == "Water"
+        )
+
+        assert (
+            final_artefact["model"].predict([[1024, 656, 575, 416, 122, 57]]) == "Water"
+        )
+
+
+def test_predict_all_function():
```

Comment:
Don't we have literally the same test in the iterator repository? Why create this code duplication?